### PR TITLE
Sentry information on update and install

### PR DESCRIPTION
**Pull request by Apollon77**
* (Apollon77) Make sure error are checked correctly (Sentry IOBROKER-…JS-CONTROLLER-1N)
* (Apollon77) Inform about sentry activation if not configured on "iobroker setup first"

### CODE REVIEW

**Comment by Apollon77** [comment]
@AlCalzone das "setup/pluginInfos.js" wäre das mit der "isReportingPlugin" methode die ich erwähnt hatte

File: lib/setup/setupSetup.js [hunk 1]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+        try {
+            ioConfig = JSON.parse(fs.readFileSync(configFile, 'utf8'));
+        } catch(err) {
+            console.error('Can not read js-controller config file. Ignore plugins defined there.');
+            ioConfig = null;
+        }
+
+        const plugins = {};
+        if (ioPackage && ioPackage.common && ioPackage.common.plugins) {
+            Object.keys(ioPackage.common.plugins).forEach(plugin => {
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:79]: I think using `Object.entries` here would fit better.

File: lib/setup/setupSetup.js [hunk 2]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+                if (ioPackage.common.plugins[plugin].enabled !== false) {
+                    plugins[plugin] = ioPackage.common.plugins[plugin];
+                }
+            });
+        }
+        if (ioConfig && ioConfig.plugins) {
+            Object.keys(ioConfig.plugins).forEach(plugin => {
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:86]: dito

File: lib/setup/setupSetup.js [hunk 3]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+                return;
+            }
+            if (systemDiag === 'none' && pluginInfos.isReportingPlugin(plugin)) {
+                // Reporting plugins respect "diag" and do not send information if diag is disabled
+                return;
+            }
+
+            let enabledState;
+            try {
+                enabledState = await states.getStateAsync('system.host.' + tools.getHostName() + '.plugins.' + plugin + '.enabled');
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:114]: template string please :)

File: lib/setup/setupSetup.js [hunk 4]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+            } catch  {
+                enabledState = null;
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:116]: unnecessary, it is already undefined.

File: lib/setup/setupSetup.js [hunk 5]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
-    function setupReady(callback) {
+    async function informAboutPlugins(systemConfig, callback) {
+        let ioPackage;
+        let ioConfig;
+
+        const configFile = tools.getConfigFileName();
+        try {
+            ioPackage = JSON.parse(fs.readFileSync(path.join(__dirname, '../../io-package.json'), 'utf8'));
+        } catch(err) {
+            console.error('Can not read js-controller io-package.json. Ignore plugins defined there.');
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:67]: I think correct english would be "Cannot" without spaces or "Can't"

File: lib/setup/setupSetup.js [hunk 6]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+        let systemLang = 'en';
+        let systemDiag = 'extended';
+        if (systemConfig && systemConfig.common) {
+            systemDiag = systemConfig.common.diag || 'extended';
+            systemLang = systemConfig.common.language || 'en';
+        }
+
+        const pluginNames = Object.keys(plugins);
+        for (let i = 0; i < pluginNames.length; i++) {
+            const plugin = pluginNames[i];
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:102]: `for (const plugin of Object.keys(plugins)) {`

File: lib/setup/setupSetup.js [hunk 7]
```diff
@@ -54,18 +56,96 @@ function Setup(options) {
...
+            if (!pluginInfos.PLUGIN_INFOS[plugin]) {
```
Comment (AlCalzone) [inline lib/setup/setupSetup.js:103]: You are accessing `pluginInfos.PLUGIN_INFOS[plugin]` ca. 5 times - I think that warrants a variable.

File: lib/setup/setupSetup.js (hunk 1 above)
Comment (Apollon77) [reply lib/setup/setupSetup.js:79]: makes sense

**Pull request merged**